BinaryValue : adding a testcase and fix for Little Endian single inde…

luked80 opened the pull request:
…x reference

Also changed the default endianness to Little Endian.
I think this is the way most rtl people naturally think of indexes.

chiggs commented on cocotb/binary.py:
```diff
@@ -69,7 +69,7 @@ class BinaryValue(object):
     _permitted_chars = "xXzZuU" + "01"  # noqa
 
-    def __init__(self, value=None, bits=None, bigEndian=True,
+    def __init__(self, value=None, bits=None, bigEndian=False,
```
@luked80 my fear about changing the default behaviour here is that it will break a lot of existing deployments.

One thing we've mooted in the past is adding a format type to `SimHandleBase` which would control the interpretation of the value (beyond `float` etc for types clearly defined by the RTL).  These could be hierarchical, such that you could globally switch to `littleEndian` by changing the DUT at the toplevel.

luked80 replied:
Yeah, that makes sense.

I've got another fix for little endian in __getitem__
Let me add that in and give you another pull request, without the change in default behaviour.

luked80 commented:
Chris,

I changed the default back to the original setting, added my fix for getitem, added a testcase and rebased onto potentialventures:master

cheers,
Luke
